docs/dialect.md

```diff
@@ -249,6 +249,17 @@ Example:
    <format-string-literal value='Standard' unicode-prefix='_utf16' />
  ```
 
+### Date Literal Support
```

Comment:
Can you updated the [SQLDialect.tdd](https://github.com/tableau/connector-plugin-sdk/blob/master/samples/components/dialects/SQLDialect.tdd) with the defaults written out in this style from Base/SQLDialect as well.